### PR TITLE
Tweak appearance of invite reason

### PR DESCRIPTION
**Pull request by jryans**
This adjusts the display of invite reasons to match design feedback.

image

image

Fixes https://github.com/vector-im/element-web/issues/16869

### CODE REVIEW

**Comment by t3chguy** [comment]
The contrast on `View message` is rough, I struggle to read it and don't have any sight issues

**Comment by jryans** [comment]
Updated to remove blur effect after discussing with @niquewoodhouse.

File: res/css/views/elements/_InviteReason.scss [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+.mx_InviteReason {
+    position: relative;
+    margin-bottom: 1em;
+
+    .mx_InviteReason_reason {
+        visibility: visible;
+    }
+
+    .mx_InviteReason_toggle {
```
Comment (t3chguy) [inline res/css/views/elements/_InviteReason.scss:25]: This doesn't really seem to be a toggle as it can only un-hide, never re-hide.

**Review by niquewoodhouse — APPROVED** [review]
Works for me.  Can iterate after feedback/shipping, eg add avatar/blur/etc.

File: src/components/views/elements/InviteReason.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    constructor(props) {
+        super(props);
+        this.state = {
+            // We hide the reason for invitation by default, since it can be a
+            // vector for spam/harassment.
+            hidden: true,
+        };
+    }
+
+    onToggleClick = () => {
```
Comment (t3chguy) [inline src/components/views/elements/InviteReason.tsx:41]: this really isn't a toggle

**Review by t3chguy — APPROVED** [review]
LGTM other than some naming nits

File: res/css/views/elements/_InviteReason.scss [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        display: none;
+        position: absolute;
+        top: 0;
+        bottom: 0;
+        left: 0;
+        right: 0;
```
Comment (gsouquet) [inline res/css/views/elements/_InviteReason.scss:31]: Sorry if you already know that, but I've only discovered this recently. There's now a shorthand for all those properties, [`inset`](https://developer.mozilla.org/en-US/docs/Web/CSS/inset)

```suggestion
        inset: 0;
```

I believe PostCSS should do its job and replace that in the final bundle

Comment (t3chguy) [reply res/css/views/elements/_InviteReason.scss:31]: Though empirically PostCSS does not translate it

image


Looks like we'd need to install https://www.npmjs.com/package/postcss-inset if we wanted that

Given lack of Safari support https://developer.mozilla.org/en-US/docs/Web/CSS/inset we should not use it without postcss support.

File: res/css/views/elements/_InviteReason.scss (hunk 1 above)
Comment (jryans) [reply res/css/views/elements/_InviteReason.scss]: Indeed, renamed to view.

File: res/css/views/elements/_InviteReason.scss (hunk 3 above)
Comment (jryans) [reply res/css/views/elements/_InviteReason.scss]: Changed to `inset`.

Comment (t3chguy) [reply res/css/views/elements/_InviteReason.scss]: @jryans probably best to revert that to not risk it breaking under Safari

**Pull request merged**